winrm/transport.py

```diff
@@ -151,13 +155,19 @@ def __init__(
     def build_session(self):
         session = requests.Session()
 
-        # allow some settings to be merged from env
-        session.trust_env = True
-        settings = session.merge_environment_settings(url=self.endpoint, proxies={}, stream=None,
-                                                      verify=None, cert=None)
+        proxies = dict()
+        if self.proxy is not None:      # pragma: no cover
+            # If there was a proxy specified then use it
+            proxies = {
+                'http': self.proxy,
+                'https': self.proxy
+            }
+
+        # Merge proxy environment variables
+        session.trust_env = self.proxy_use_env
```

Comment:
IIRC, `session.trust_env` applies to everything; anything that unsets it this way will prevent *all* the env stuff from working, not just proxies (eg CA cert bundle location, auth settings, etc). `proxy_ignore_env` probably needs to be a more global `config_ignore_env` unless you're going to re-create what requests is pulling from the env and pass it in here. It's been a long time since I wrote that code though, so I don't remember exactly how requests manages that for a session (had to go spelunking in the requests code to figure it out).